src/ai/ai_instance.cpp

```diff
@@ -258,7 +258,7 @@ void CcAI(const CommandCost &result, TileIndex tile, uint32 p1, uint32 p2)
 	 * when the company does not exist anymore.
 	 */
 	const Company *c = Company::GetIfValid(_current_company);
-	if (c == NULL || c->ai_instance == NULL) return;
+	if (c == NULL || c->ai_instance == NULL || !c->ai_instance->IsWaiting()) return;
```

Comment:
This doesn't add any checks to make sure it's the same ai_instance as that which called the command in the first place.